### PR TITLE
Log an uncaught exception before it crashes the app.

### PR DESCRIPTION
**Pull request by jotomo**
According to https://fabric.io/kits/android/crashlytics/features (Step 3, 2nd paragraph), registering an UncaughtExceptionHandler before Fabric is set up does not interfere with Fabric.

This will help getting access to stracktraces (via the log) which otherwise are only visible via logcat.

### CODE REVIEW

**Comment by MilosKozak** [comment]
should not we send it to firebase by calling `FabricPrivacy.logException(...)`?

**Comment by jotomo** [comment]
Fabric already does that, it also registers an uncaught exception handler. However, the issues with starting foreground services didn't create Fabric reports. (When Android asks the JVM to shut down, it disallows creating new threads. But since Fabric creates network traffic it either must have a thread on standby or create a new background thread since network I/O on the main thread is not allowed. So my speculation is that Fabric is unable to report in that scenario due to this).
In short: this handles the case where Fabric doesn't report anything but makes sure something goes into the AAPS' logs, rather than just Android's system log.

**Comment by MilosKozak** [comment]
great!

**Comment by MilosKozak** [comment]
let's try it
